downstairs/src/lib.rs

```diff
@@ -3674,7 +3719,10 @@ mod test {
             .iter()
             .all(|dep| work.completed.is_complete(*dep)));
 
-        if matches!(job, IOop::Flush { .. }) {
+        // Flushes and barriers both guarantee that no future jobs will depend
+        // on jobs that preceded them, so we reset the completed jobs list to
+        // only include their job id.
+        if matches!(job, IOop::Flush { .. } | IOop::Barrier { .. }) {
```

Comment:
This means the lowest job in `work.completed` will always be either a flush or a barrier?

Reply:
Yup, exactly